MirandaRoland/StoreProject/src/Report.java

```diff
@@ -0,0 +1,111 @@
+import java.io.IOException;
+import javax.swing.JLabel;
+import javax.swing.JOptionPane;
+import javax.swing.JPanel;
+import javax.swing.JTable;
+
+
+public final class Report implements displayable{
+    private Package[] list;
+    private String header;
+    private String bodyText;
+    private double TotalAsset;
+    private double costOFissuedItems;
+    private double costOFdispatched;
+    private final Object[][] comparePackagesTable;
+
+    public Report(Package[] list,String header,String bodyText){
+        this.list=list;
+        this.header=header;
+        this.bodyText=bodyText;
+        comparePackagesTable = new Object[list.length+1][2];//two columns, plus one row for columns titles
+        comparePackagesTable[0][0]="code";
+        comparePackagesTable[0][1]="percentage";
+        calculate();
+    }
+    @Override
+    public void display(){
+        if(GUI.DEFAULT_UI!=userInterface.GRAPHICAL){
+            gShow();
+        }else{
+            tShow();
+        }
+    }
+    @Override
+    public String toString(){
+        String temp=header+"\r\n"+bodyText+"\r\n"+
+                "the following table shows property model with its\r\n" +
+                "corresponding percentage of its price compared with\r\n" +
+                "the total asset of the store. note that the percentage\r\n" +
+                "is rounded.\r\n"+
+                "____________________________\r\n";
+        temp+=String.format("%15s|%5s %n",comparePackagesTable[0][0],comparePackagesTable[0][1]);
+        for(int i=1,listWithColumen=list.length+1;i<listWithColumen;i++){
+            temp+=String.format("%15s|%10f %n",comparePackagesTable[i][0],(double)comparePackagesTable[i][1]);
+        }
+        temp+="____________________________\r\n"
+                +"total asset of the store is "+TotalAsset+" $\r\n"
+                + "the issued items will cost "+costOFissuedItems+" $\r\n"
+                + "the disptched items cost "+costOFdispatched+" $\r\n\r\n"
+                + "this is generated using store management program.";
+        return temp;
+    }
+
+    public double getTotalAsset(){
+        return TotalAsset;
+    }
+    public void calculate(){
+        TotalAsset=0;
+        costOFdispatched=0;
+        costOFissuedItems=0;
+        for (Package l : list) {
```

Comment:
Please do not add variable names that are only one letter length, is not readable.